crypto/digestible/derive/src/lib.rs

```diff
@@ -547,6 +549,103 @@ fn try_digestible_enum(
     Ok(expanded.into())
 }
 
+fn try_digestible_enum_transparent(
+    ident: &Ident,
+    generics: &Generics,
+    variant_data: &DataEnum,
+) -> Result<TokenStream, &'static str> {
```

Comment:
What does this do? It looks to me like this is trying to digest the VersionedAmount enum in a zero knowledge way - is that correct? I know this is a research PR but, if you find this direction a worthy path to pursue, def document this one!

Reply:
yeah if it looks like this is working out, i think we need to pull out the digestible-derive changes into their own PR to ease review

Reply:
It does indeed look like it's working out